### PR TITLE
Fix project_*_containers C++ and tests

### LINKED ISSUE #1001: False positives on project template tests
```
test 1
      Start  1: project_root_containers

1: Test command: /tmp/fairsoft_ctest.1X6EpE/spSF4V/FairRoot/build/templates/test_project_root_containers.sh "PROPERTIES" "TIMEOUT" "300" "PASS_REGULAR_EXPRESSION" "Test successful."
1: Test timeout computed to be: 1500
1: -- The C compiler identification is GNU 9.3.0
1: -- The CXX compiler identification is GNU 9.3.0
1: -- Check for working C compiler: /usr/bin/cc
1: -- Check for working C compiler: /usr/bin/cc -- works
1: -- Detecting C compiler ABI info
1: -- Detecting C compiler ABI info - done
1: -- Detecting C compile features
1: -- Detecting C compile features - done
1: -- Check for working CXX compiler: /usr/bin/c++
1: -- Check for working CXX compiler: /usr/bin/c++ -- works
1: -- Detecting CXX compiler ABI info
1: -- Detecting CXX compiler ABI info - done
1: -- Detecting CXX compile features
1: -- Detecting CXX compile features - done
1: -- Setting FairRoot environment:
1: --   FairRoot prefix            : /tmp/fairsoft_ctest.1X6EpE/spSF4V/FairRoot/install
1: --   FairRoot Library directory : /tmp/fairsoft_ctest.1X6EpE/spSF4V/FairRoot/install/lib
1: --   FairRoot Include path      : /tmp/fairsoft_ctest.1X6EpE/spSF4V/FairRoot/install/include
1: --   FairRoot Cmake Modules     : /tmp/fairsoft_ctest.1X6EpE/spSF4V/FairRoot/install/share/fairbase/cmake
1: Set BuildType DEBUG
1: -- fairsoft-config found
1: Files "/tmp/fairsoft_ctest.1X6EpE/FairSoft/spack/lib/spack/env/cc" to "/usr/bin/x86_64-linux-gnu-g++-9" are different.
1: -- C compiler used for FairSoft installation:  /tmp/fairsoft_ctest.1X6EpE/FairSoft/spack/lib/spack/env/cc
1: -- C compiler used now:  /usr/bin/x86_64-linux-gnu-gcc-9
1: -- CXX compiler used for FairSoft installation:  /tmp/fairsoft_ctest.1X6EpE/FairSoft/spack/lib/spack/env/cc
1: -- CXX compiler used now:  /usr/bin/x86_64-linux-gnu-g++-9
1: -- The compiler during the compilation of FairSoft is different from the current one.
1: -- The error was silenced by the usage of -DUSE_DIFFERENT_COMPILER=TRUE
1: --- Found a Linux system
1: --- Found GNU compiler collection
1: -- ROOT::Core target added by hand.
1: -- ROOT::RIO target added by hand.
1: -- ROOT::Tree target added by hand.
1: -- ROOT::Rint target added by hand.
1: -- ROOT::Physics target added by hand.
1: -- ROOT::MathCore target added by hand.
1: -- ROOT::Thread target added by hand.
1: -- ROOT::Geom target added by hand.
1: -- ROOT::EG target added by hand.
1: -- Looking for HepMC... - found /tmp/fairsoft_ctest.1X6EpE/spSF4V/fairsoft/lib
1: -- Performing Test _HAS_CXX11_FLAG
1: -- Performing Test _HAS_CXX11_FLAG - Success
1: -- Checking C++11 support for "__func__"
1: -- Checking C++11 support for "__func__": works
1: -- Checking C++11 support for "auto"
1: -- Checking C++11 support for "auto": works
1: -- Checking C++11 support for "auto_ret_type"
1: -- Checking C++11 support for "auto_ret_type": works
1: -- Checking C++11 support for "class_override_final"
1: -- Checking C++11 support for "class_override_final": works
1: -- Checking C++11 support for "constexpr"
1: -- Checking C++11 support for "constexpr": works
1: -- Checking C++11 support for "constructor-delegation"
1: -- Checking C++11 support for "constructor-delegation": works
1: -- Checking C++11 support for "cstdint"
1: -- Checking C++11 support for "cstdint": works
1: -- Checking C++11 support for "decltype"
1: -- Checking C++11 support for "decltype": works
1: -- Checking C++11 support for "initializer_list"
1: -- Checking C++11 support for "initializer_list": works
1: -- Checking C++11 support for "lambda"
1: -- Checking C++11 support for "lambda": works
1: -- Checking C++11 support for "long_long"
1: -- Checking C++11 support for "long_long": works
1: -- Checking C++11 support for "nullptr"
1: -- Checking C++11 support for "nullptr": works
1: -- Checking C++11 support for "regex"
1: -- Checking C++11 support for "regex": works
1: -- Checking C++11 support for "rvalue-references"
1: -- Checking C++11 support for "rvalue-references": works
1: -- Checking C++11 support for "sizeof_member"
1: -- Checking C++11 support for "sizeof_member": works
1: -- Checking C++11 support for "static_assert"
1: -- Checking C++11 support for "static_assert": works
1: -- Checking C++11 support for "variadic_templates"
1: -- Checking C++11 support for "variadic_templates": works
1: -- Checking C++11 support for "sharedpointer"
1: -- Checking C++11 support for "sharedpointer": works
1: -- Checking C++11 support for "funcptr-to-lambda-conversion"
1: -- Checking C++11 support for "funcptr-to-lambda-conversion": works
1: --   
1: --   GLOBAL CXX FLAGS   -std=c++11
1: --   
1: --   BUILD TYPE         CXX FLAGS
1: --   RELWITHDEBINFO     -O2 -g -Wshadow
1: --   NIGHTLY            -O0 -g -Wshadow
1: --   TEST               -O2 -g -Wshadow
1: --   RELEASE            -O2 -Wshadow 
1: -- * DEBUG              -g -Wshadow 
1: --   DEBUGFULL          -g3 -fno-inline -Wnon-virtual-dtor -Wno-long-long -ansi -Wundef -Wcast-align -Wchar-subscripts -Wall -W -Wpointer-arith -Wformat-security -fno-exceptions -fno-check-new -fno-common -fexceptions
1: --   PROFILE            -g3 -fno-inline -ftest-coverage -fprofile-arcs -Wshadow -Wall -Wextra -Wunused-variable
1: --   ARRAY_CHECK        -g3 -fno-inline -ftest-coverage -fprofile-arcs -fstack-protector
1: --   
1: --   (Change the build type with -DCMAKE_BUILD_TYPE=...)
1: -- Configuring done
1: -- Generating done
1: -- Build files have been written to: /tmp/tmp.rZuuegIfxp/project_root_containers/build
1: Scanning dependencies of target svnheader
1: -- Found Git: /usr/bin/git (found version "2.25.1") 
1: [  0%] Built target svnheader
1: [  3%] Generating G__TstDataDict.cxx, G__TstDataDict_rdict.pcm, ../lib/libTstData.rootmap
1: In file included from input_line_12:3:
1: In file included from /tmp/tmp.rZuuegIfxp/project_root_containers/TstData/TstStack.h:35:
1: /tmp/fairsoft_ctest.1X6EpE/spSF4V/FairRoot/install/include/FairGenericStack.h:23:10: fatal error: 'TMCProcess.h' file not found
1: #include <TMCProcess.h>        // for TMCProcess
1:          ^~~~~~~~~~~~~~
1: Error: /tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/bin/rootcint: compilation failure (/tmp/tmp.rZuuegIfxp/project_root_containers/build/TstData/G__TstDataDictcf245a372b_dictUmbrella.h)
1: make[2]: *** [TstData/CMakeFiles/TstData.dir/build.make:64: TstData/G__TstDataDict.cxx] Error 1
1: make[1]: *** [CMakeFiles/Makefile2:969: TstData/CMakeFiles/TstData.dir/all] Error 2
1: make: *** [Makefile:141: all] Error 2
 1/54 Test  #1: project_root_containers ................   Passed    9.41 sec
```

Test is **passed** despite the compilation error. The same happens for `project_stl_containers` test.

In this particular case, we also need to update the project templates to be able to deal with external VMC.

### PR DESCRIPTION
**Pull request by ChristianTackeGSI**
Should fix #1001

C++ Constructors in project_stl_containers
------------------------------------------

project_stl_containers wants an event id in its Point class, so pass it around.

Build System for project_*_containers
-------------------------------------
* We need to find_package2() for VMC and fmt.
* Let's start to use ROOT:: qualified targets.

add_test Syntax Fixes
---------------------
There's a simple syntax
```cmake
add_test(test_name command arg1 arg2)
```
This syntax captures *everything* after the command as arguments.

Let's use the better syntax:
```cmake
add_test(NAME test_name COMMAND command arg1 arg2)
```

Also the project_*_containers tests added PROPERTIES here. That doesn't work at all.  One should use set_tests_properties().

---

Checklist:

* [X] Rebased against `dev` branch
* [X] My name is in the resp. CONTRIBUTORS/AUTHORS file
* [X] Followed [the seven rules of great commit messages](https://chris.beams.io/posts/git-commit/#seven-rules)

### CODE REVIEW

File: templates/project_root_containers/CMakeLists.txt [hunk 1]
```diff
@@ -96,7 +96,21 @@ if(SIMPATH)
...
 
 find_package2(PUBLIC ROOT  VERSION 6.10.00  REQUIRED)
+include(ROOTMacros)
+if(ROOT_vmc_FOUND)
+  set(VMCLIB ROOT::VMC)
+else()
+  find_package2(PUBLIC VMC REQUIRED)
+  set(VMCLIB VMCLibrary)
+endif()
+message(STATUS "Using VMC: ${VMCLIB}")
```
Comment (dennisklein) [inline templates/project_root_containers/CMakeLists.txt:106]:
```suggestion
```

Let's drop this or put it on different verbosity that is not printed by default.

File: templates/project_root_containers/CMakeLists.txt [hunk 2]
```diff
@@ -96,7 +96,21 @@ if(SIMPATH)
 endif()
 
 find_package2(PUBLIC ROOT  VERSION 6.10.00  REQUIRED)
```
Comment (dennisklein) [inline templates/project_root_containers/CMakeLists.txt:98]: Let's either drop the version requirement or update it.

File: templates/project_stl_containers/NewDetector/NewDetectorPoint.h [hunk 3]
```diff
@@ -27,6 +27,7 @@ class NewDetectorPoint
      *@param eLoss    Energy deposit [GeV]
      **/
     NewDetectorPoint(Int_t trackID,
+                     uint32_t evtId,
```
Comment (dennisklein) [inline templates/project_stl_containers/NewDetector/NewDetectorPoint.h:30]: @kresan  Can you verify if this is a meaningful/correct/intended way of passing/initializing the event id (why is this difference between the stl and root container templates?)

**Review by dennisklein — CHANGES_REQUESTED** [review]
Can you please split the C++ changes into a separate commit, thx!

File: templates/project_root_containers/CMakeLists.txt (hunk 2 above)
Comment (ChristianTackeGSI) [reply templates/project_root_containers/CMakeLists.txt]: Dropped for now.

I think, the version requirement here should reflect which version of ROOT the project_*_containers need themselves.

File: templates/project_root_containers/CMakeLists.txt (hunk 1 above)
Comment (ChristianTackeGSI) [reply templates/project_root_containers/CMakeLists.txt]: dropped.

**Comment by ChristianTackeGSI** [comment]
> Can you please split the C++ changes into a separate commit, thx!

done.

All other things should be addressed too.

File: templates/project_stl_containers/NewDetector/NewDetectorPoint.h (hunk 3 above)
Comment (kresan) [reply templates/project_stl_containers/NewDetector/NewDetectorPoint.h:30]: EventID is not needed in the NewDetectorPoint class, please remove it from both templates. Ideally, project_stl_containers should not use ROOT types.

Comment (ChristianTackeGSI) [reply templates/project_stl_containers/NewDetector/NewDetectorPoint.h]: Removed `fEventId` and all users from `project_stl_containers`. Please recheck now.

**Review by kresan — APPROVED** [review]
Looks good. Do the templates compile successfully? I guess so, since "All checks have passed".

**Comment by ChristianTackeGSI** [comment]
Yes, both compile. With a few warnings.
Goal of this PR:
* Let the tests compile both projects (was already working)
* AND let the tests fail, iff the compile (the tests always succeeded before)
* Fix all compile errors in the projects, so that the tests succeed.

Taking a deeper look at the projects and cleaning them up (fixing warnings, porting them to newer ideas) is a new task.

**Review by dennisklein — APPROVED** [review]

**Pull request merged**